client/extensions/woocommerce/components/product-search/row.js

```diff
@@ -0,0 +1,280 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { filter, find, get, intersection, noop, reduce, uniqBy, values } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Count from 'components/count';
+import { fetchProductVariations } from 'woocommerce/state/sites/product-variations/actions';
+import formatCurrency from 'lib/format-currency';
+import formattedVariationName from 'woocommerce/lib/formatted-variation-name';
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+import FormCheckbox from 'components/forms/form-checkbox';
+import { getPaymentCurrencySettings } from 'woocommerce/state/sites/settings/general/selectors';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { getVariationsForProduct } from 'woocommerce/state/sites/product-variations/selectors';
+import { isProductSelected, isVariableProduct } from './utils';
+import ProductVariations from './variations';
+
+class ProductSearchRow extends Component {
+	static propTypes = {
+		onChange: PropTypes.func,
+		product: PropTypes.object.isRequired,
+		singular: PropTypes.bool,
+		value: PropTypes.oneOfType( [ PropTypes.number, PropTypes.array ] ),
+	};
+
+	static defaultProps = {
+		onChange: noop,
+		singular: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			showForm: false,
+			variations: [],
+		};
+	}
+
+	componentDidMount() {
+		const { siteId, product, productId } = this.props;
+
+		if ( siteId && productId ) {
+			if ( ! product || 'variable' !== product.type ) {
+				return;
+			}
+			this.props.fetchProductVariations( siteId, productId );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		// Short out if we know the new product doesn't need variations
+		if ( ! newProps.product || 'variable' !== newProps.product.type ) {
+			return;
+		}
+
+		const { siteId: oldSiteId, productId: oldProductId } = this.props;
+		const { siteId: newSiteId, productId: newProductId } = newProps;
+
+		if ( oldSiteId !== newSiteId || oldProductId !== newProductId ) {
+			this.props.fetchProducts( newSiteId, newProductId );
+		}
+
+		const selectedIds = intersection( newProps.value, newProps.product.variations );
+		const selectedVariations = selectedIds.map( id => find( newProps.variations, { id } ) );
+		this.state = {
```

Comment:
This should use `setState` instead. It's causing a console warning.

Reply:
hah, bet that's new with the React upgrade, hadn't seen that the rest of the time I was working on this